packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -578,78 +586,99 @@ async function createComponentTreeInternal({
+        </Segment>
+      </React.Fragment>,
+      loadingData,
+    ]
   } else {
     // For layouts we just render the component
-    segmentElement = (
-      <>
+    return [
+      actualSegment,
+      parallelRouteCacheNodeSeedData,
+      // It is critical that this tree render something other than `null` because the client router uses
+      // null to represent an lazy hole. The current implementation satisfies this because the two inner slots
+      // ensure there is a fragment even if both slots render null. If we ever refactor this to only render the component
+      // or similar we need to ensure there is a fragment. Long term we should move to using a Symbol to communicate
+      // a lazy hole rather than null
+      <Segment
+        key={cacheNodeKey}
+        isStaticGeneration={isStaticGeneration}
+        ready={getMetadataReady}
+      >
         {layerAssets}
         <Component {...props} />
-      </>
-    )
+      </Segment>,
+      loadingData,
+    ]
   }
-
-  return [
-    actualSegment,
-    parallelRouteCacheNodeSeedData,
-    <React.Fragment key={cacheNodeKey}>
-      {segmentElement}
-      {/* This null is currently critical. The wrapped Component can render null and if there was not fragment
-            surrounding it this would look like a pending tree data state on the client which will cause an error
-            and break the app. Long-term we need to move away from using null as a partial tree identifier since it
-            is a valid return type for the components we wrap. Once we make this change we can safely remove the
-            fragment. The reason the extra null here is required is that fragments which only have 1 child are elided.
-            If the Component above renders null the actual tree data will look like `[null, null]`. If we remove the extra
-            null it will look like `null` (the array is elided) and this is what confuses the client router.
-            TODO-APP update router to use a Symbol for partial tree detection */}
-      {null}
-    </React.Fragment>,
-    loadingData,
-  ]
 }
 
 async function MetadataOutlet({
-  getReady,
+  ready,
 }: {
-  getReady: () => Promise<void> & { status?: string; value?: unknown }
+  ready: () => Promise<void> & { status?: string; value?: unknown }
 }) {
-  const ready = getReady()
+  const r = ready()
   // We actually expect this to be an instrumented promise and once this file is properly
   // moved to the RSC module graph we can switch to using React.use for this synchronous unwrapping.
   // The synchronous unwrapping will become important with dynamic IO since we want to resolve metadata
   // before anything dynamic can be triggered
-  if (ready.status === 'rejected') {
-    throw ready.value
-  } else if (ready.status !== 'fulfilled') {
-    await ready
+  if (r.status === 'rejected') {
+    throw r.value
+  } else if (r.status !== 'fulfilled') {
+    // returning here is like awaiting the ready promise
```

Comment:
what does this comment mean? AFAICT we are literally awaiting the ready promise, what's "like" about it? and what does "returning" mean here?

Reply:
I think I left this comment behind from an abandoned refactor. I’ll remove. In fact the comment above also only makes sense if this was a client component. Since you wouldn’t typically `use` a promise in a server component